api/app/Traits/HasLocalizedEnums.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Traits;
+
+trait HasLocalizedEnums
+{
+    protected function localizeEnum(?string $value, $enum)
+    {
+        if (! $value || is_object($enum) || ! method_exists($enum, 'localizedString')) {
+            return $value;
+        }
+
+        return [
```

Comment:
Is this a fallback that, ideally, should not be hit?  Worth logging an error?

Reply:
The inverse, this is the expected result. The if statement guards against converting non-localized enums. I'll add a log to that, just wondering if you have any input as the the severity we should log, I was thinking warning?

Reply:
I'm not super happy with the guard, but we are now logging misuse :sweat_smile: 

[log attempt to localize non-localizable enums](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10701/commits/1140f746cca83ff20d35e53c4144cab1285e451a)